ci/nix/Dockerfile

```diff
@@ -0,0 +1,100 @@
+# This image is only for extracting and cleaning up Android SDK
+FROM ubuntu:18.04 AS android_sdk
+
+ARG ANDROID_SDK_VERSION
+ARG SDK_PLATFORM_VERSION
+ARG SDK_BUILD_TOOLS_VERSION
+
+# names of packages to install with sdkmanager
+ENV SDK_PACKAGES emulator patcher;v4 \
+                 platforms;android-26 \ 
+                 platforms;${SDK_PLATFORM_VERSION} \
+                 build-tools;${SDK_BUILD_TOOLS_VERSION} \
+                 extras;android;m2repository extras;google;m2repository
+
+ENV PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
+    LANG=en_US.UTF-8 \
+    LC_ALL=en_US.UTF-8 \
+    LANGUAGE=en_US.UTF-8
+
+RUN apt-get update && apt-get -q -y --no-install-recommends install software-properties-common \
+ && add-apt-repository -y ppa:webupd8team/java \
+ && echo debconf shared/accepted-oracle-license-v1-1 select true | debconf-set-selections \
+ && echo debconf shared/accepted-oracle-license-v1-1 seen true | debconf-set-selections \
+ && apt-get update \
+ && DEBIAN_FRONTEND=noninteractive apt-get -q -y --no-install-recommends install \
+      wget unzip locales ca-certificates oracle-java8-installer oracle-java8-set-default \
+ && locale-gen ${LANG} \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /usr/share/man
+
+# install Android SDK
+COPY sdk-tools-linux-${ANDROID_SDK_VERSION}.zip /tmp/sdk-tools-linux.zip
+RUN unzip -q /tmp/sdk-tools-linux.zip -d /usr/lib/android-sdk \
+ && mkdir ~/.android && touch ~/.android/repositories.cfg
+RUN /usr/lib/android-sdk/tools/bin/sdkmanager --licenses
+RUN for PKG in ${SDK_PACKAGES}; do \
+      yes | /usr/lib/android-sdk/tools/bin/sdkmanager --install "${PKG}" > /dev/null; done \
+ && chmod 777 -R /usr/lib/android-sdk
+# cleanup SDK
+RUN cd /usr/lib/android-sdk \
+ && rm -fr extras emulator ndk-bundle/toolchains/{aarch64-linux-android-4.9,x86-4.9,x86_64-4.9,llvm}
+
+################################################################################
+FROM ubuntu:18.04
+ENV LANG=en_US.UTF-8 \
+    LC_ALL=en_US.UTF-8 \
+    LANGUAGE=en_US.UTF-8
+
+ARG NIX_VERSION
+ARG DEPS_HASH
```

Comment:
Do we do this just so it's part of image env for documentation sake? Because we're not using it for anything here.

Reply:
I should add a comment. The goal is for certain dependencies which are not really directly used in the building of the image to still cause a new image to be generated (think StatusImWindowsBaseImage.zip which is now downloaded by Nix)

Reply:
Ah, makes sense, clever. A commend would be good.